include/stl2/detail/iterator/concepts.hpp

```diff
@@ -186,13 +186,14 @@ STL2_OPEN_NAMESPACE {
 	///////////////////////////////////////////////////////////////////////////
 	// Writable [iterators.writable]
 	// Not to spec, but getting closer.
-	// See https://github.com/ericniebler/stl2/issues/240
-	// Additional changes from P0547
+	// See https://github.com/ericniebler/stl2/issues/339
+	// and https://github.com/ericniebler/stl2/issues/381
 	//
 	template <class Out, class R>
 	concept bool Writable() {
 		return detail::Dereferenceable<Out> && requires(Out& o, R&& r) {
 			(void)(*o = (R&&)r);
+			(void)(const_cast<add_const_t<reference_t<Out>>&&>(*o) = (R&&)r);
```

Comment:
I'd prefer `const` here to `add_const_t`, IIRC they're equivalent for referenceable types and `Dereferenceable` guarantees that `reference_t<Out>` is a referenceable type.